state/pruning_state.py

```diff
@@ -98,7 +98,8 @@ def generate_state_proof(self, key: bytes, root=None, serialize=False):
 
     @staticmethod
     def verify_state_proof(root, key, value, proof_nodes, serialized=False):
-        return Trie.verify_spv_proof(root, key, rlp_encode([value]),
+        encoded_value = rlp_encode([value]) if value is not None else b''
+        return Trie.verify_spv_proof(root, key, encoded_value,
```

Comment:
Can you please add a test for verify_state_proof when value is None?

Reply:
Added